src/main/java/com/rovio/ingest/TaskDataWriter.java

```diff
@@ -214,9 +212,9 @@ private Map<String, Object> parse(InternalRow record) {
             } else {
                 DataType sqlType = field.getSqlType();
                 Object value = record.get(field.getOrdinal(), sqlType);
-                if (sqlType == DataTypes.StringType && value instanceof UTF8String) {
-                    // Convert to String as Spark return UTF8String which is not compatible with Druid sketches.
-                    value = new String(((UTF8String) value).getBytes(), StandardCharsets.UTF_8);
+                if (value != null && segmentSpec.getComplexMetricColumns().contains(columnName) && sqlType == DataTypes.StringType) {
+                    // Convert to Java String as Spark return UTF8String which is not compatible with Druid sketches.
+                    value = value.toString();
```

Comment:
What is this change about (this line)?

Reply:
value here when the field is StringType is of instance https://spark.apache.org/docs/1.4.0/api/java/org/apache/spark/sql/types/UTF8String.html.
The change here is coerce it to Java string 
https://spark.apache.org/docs/1.4.0/api/java/org/apache/spark/sql/types/UTF8String.html#toString()

Reply:
calling toString() on UTF8String is same as the previous code.
I changed to toString() to avoid checking if object is instance of UTF8String & also type casting it.